jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TraceContextCodecTest.java

```diff
@@ -160,10 +164,31 @@ public void testEmptyTraceStateNotPropagated() {
     assertEquals(EXAMPLE_TRACE_PARENT, injectCarrier.get(TRACE_PARENT));
   }
 
+  @Test
+  public void testDebugIdWithoutTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertTrue(spanContext.isDebug());
+  }
+
+  @Test
+  public void testDebugIdWithTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(TRACE_PARENT, EXAMPLE_TRACE_PARENT);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertTrue(spanContext.isDebug());
```

Comment:
Same as above.

What we really want to test is whether the tracer will be starting spans with debug flag when given this span context as a parent. But such test should be in the TracerTests.